crates/data_components/src/arrow/write.rs

```
@@ -177,7 +180,19 @@ impl TableProvider for MemTable {
             ));
         }
 
-        let sink = Arc::new(MemSink::new(self.batches.clone(), overwrite));
+        // If the table has a primary key, we need to know it to handle `InsertOp::Replace`.
+        let mut primary_key = None;
+        if let Some(constraints) = self.constraints() {
```

This should return an error on constraints we don't handle, i.e. unique